presto-plugin-toolkit/src/main/java/io/prestosql/plugin/base/security/SchemaAccessControlRule.java

```diff
@@ -24,26 +24,42 @@
 public class SchemaAccessControlRule
 {
     private final boolean owner;
+    private final Optional<Boolean> visible;
     private final Optional<Pattern> userRegex;
     private final Optional<Pattern> schemaRegex;
 
     @JsonCreator
     public SchemaAccessControlRule(
             @JsonProperty("owner") boolean owner,
+            @JsonProperty("visible") Optional<Boolean> visible,
```

Comment:
How does this relate to `allow` in `CatalogAccessControlRule` introduced in 138394202ab9cbc6162444b7d335251c87431418?

Reply:
Good question. `allow` propagates and so it is used to allow user to do certain thing in given catalog, `visible` does not allow to do anything in given schema other than to see that it exists.

`visible=true` sounds similar to `allow=read-only`, but `visible=true` does not allow to see tables in give schema.

It would be nice to have `allow` implemented on schema level (and so on table level), but I think we would need to have new state like `allow=visible` and it would also require implementing other states for `allow` which sounds like a bit out of the scope of this PR. However, introducing something that we would like to replace in the near future does not sound like a good idea either.

Any thoughts?

Reply:
> `visible` does not allow to do anything in given schema other than to see that it exists.

Why do we need this? What problem does it solve?

Why can't we show schema iff user can do something in it? Since we're file-based, there eg is no cost incurred of talking to remote services.

Reply:
I agree. The best would be to have `allow` implemented for `SchemaAccessControlRule` in the same way as it is implemented for `CatalogAccessControlRule`. However this could be out of the scope.

@l0yu Instead of adding `visible` please make schema visible if user is an owner of schema or there is a table rule that potentially would allow access to a table (maybe not existing one, because we are not going to list tables in access control) in that schema.

Reply:
@findepi @kokosing 
I am not disagree on iterate over all tables, but from user experience (rule in file is to be composed/edit by admin or operators)
If changing visibility of schema, user have to look through all table rules
and edit one by one in file, it's error-prone
If we define schema privilege, it is just one line change in config file.

Reply:
> If changing visibility of schema, user have to look through all table rules
and edit one by one in file,

That is correct, to handle this case:
>  The best would be to have `allow` implemented for `SchemaAccessControlRule` in the same way as it is implemented for `CatalogAccessControlRule`.

Reply:
@kokosing do you mean replace ` visible ` boolean with similar implementation as
https://github.com/prestosql/presto/blob/master/presto-plugin-toolkit/src/main/java/io/prestosql/plugin/base/security/CatalogAccessControlRule.java#L55